app/src/main/java/com/jerboa/ui/components/post/PostActivity.kt

```diff
@@ -305,6 +311,17 @@ fun PostActivity(
                     is ApiState.Failure -> ApiErrorText(postRes.msg)
                     is ApiState.Success -> {
                         val postView = postRes.data.post_view
+                        if (!account.isAnon() && !postView.read) {
+                            appState.addReturn(PostViewReturn.POST_VIEW, postView.copy(read = true))
+                            postViewModel.markPostAsRead(
+                                MarkPostAsRead(
+                                    post_id = postView.post.id,
+                                    read = true,
+                                    auth = account.jwt,
+                                ),
+                                appState,
+                            )
```

Comment:
Well this isn't needed anymore and you can remove that function too as I don't think we will ever need it now.

Reply:
Okay, I've removed it and it will only add the consumable return when the post has successfully loaded.